### ISSUE TITLE
After the end of Chaos Castle, the ancient equipment obtained cannot be saved.

### ISSUE THREAD

**Issue by sd3420625** [MUnique/OpenMU#448]
**Describe the bug**
After many experiments, although the ancient equipment rewarded by Chaos Castle can be picked up, it cannot be traded to others, and the person who picks it up will not be able to pop up the trading interface. And when the picker logs in again, he will return to the Chaos Castle map, and the rewarded ancient equipment will disappear.

**To Reproduce**
1. Go to 'Chaos Castle'
2. Finish the mini game
3. Pick up the ancient equipment
4. Return to the mainland
5. Trade other people or Log in to the game again

**Expected behavior**
Characters can save correctly while holding reward equipment and can trade ancient equipment normally.

**Desktop (please complete the following information):**
 - OS: [Ubuntu22.04]
 - mode: docker-build, all-in-one-traefik

**Comment by sven-n** [comment]
Thank you for the detailed error report. I can reproduce the issue this is the error message in the log, after trying to trade another player:
```
2024-08-14 12:54:27.453 +02:00 [Error] [MUnique.OpenMU.GameLogic.Player] Error while processing the message: C1 04 37 01
System.InvalidOperationException: The instance of entity type 'ItemItemOfItemSet' cannot be tracked because another instance with the same key value for {'ItemId', 'ItemOfItemSetId'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.NavigationFixer.NavigationCollectionChanged(InternalEntityEntry entry, INavigationBase navigationBase, IEnumerable`1 added, IEnumerable`1 removed)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.ChangeDetector.DetectNavigationChange(InternalEntityEntry entry, INavigationBase navigationBase)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.ChangeDetector.LocalDetectChanges(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.ChangeDetector.DetectChanges(IStateManager stateManager)
   at Microsoft.EntityFrameworkCore.ChangeTracking.ChangeTracker.DetectChanges()
   at Microsoft.EntityFrameworkCore.DbContext.TryDetectChanges()
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at MUnique.OpenMU.Persistence.EntityFramework.EntityFrameworkContextBase.SaveChangesAsync(CancellationToken cancellationToken) in C:\Users\Sven\Documents\GitHub\OpenMU_2\src\Persistence\EntityFramework\EntityFrameworkContextBase.cs:line 86
   at MUnique.OpenMU.GameLogic.PlayerActions.Trade.TradeAcceptAction.OpenTradeAsync(ITrader trader) in C:\Users\Sven\Documents\GitHub\OpenMU_2\src\GameLogic\PlayerActions\Trade\TradeAcceptAction.cs:line 58
   at MUnique.OpenMU.GameLogic.PlayerActions.Trade.TradeAcceptAction.HandleTradeAcceptAsync(ITrader tradeAccepter, Boolean accept) in C:\Users\Sven\Documents\GitHub\OpenMU_2\src\GameLogic\PlayerActions\Trade\TradeAcceptAction.cs:line 34
   at MUnique.OpenMU.GameServer.MessageHandler.Trade.TradeAcceptHandlerPlugIn.HandlePacketAsync(Player player, Memory`1 packet) in C:\Users\Sven\Documents\GitHub\OpenMU_2\src\GameServer\MessageHandler\Trade\TradeAcceptHandlerPlugIn.cs:line 33
   at MUnique.OpenMU.GameServer.MessageHandler.PacketHandlerPlugInContainer`1.HandlePacketAsync(Player player, Memory`1 packet, THandler handler) in C:\Users\Sven\Documents\GitHub\OpenMU_2\src\GameServer\MessageHandler\PacketHandlerPlugInContainer.cs:line 122
   at MUnique.OpenMU.GameServer.MessageHandler.PacketHandlerPlugInContainer`1.HandlePacketAsync(RemotePlayer player, Memory`1 packet) in C:\Users\Sven\Documents\GitHub\OpenMU_2\src\GameServer\MessageHandler\PacketHandlerPlugInContainer.cs:line 73
   at MUnique.OpenMU.GameServer.RemoteView.RemotePlayer.PacketReceivedAsync(ReadOnlySequence`1 sequence) in C:\Users\Sven\Documents\GitHub\OpenMU_2\src\GameServer\RemoteView\RemotePlayer.cs:line 143
```

**Comment by sven-n** [comment]
Should be fixed with #453 and #454
